### ISSUE TITLE
IPFIX - Unable to parse PCAP file.

### ISSUE THREAD

**Issue by lingpri** [Graylog2/graylog-plugin-integrations#371]
## Description
As a developer ,research on why the "ixflow.pcap" cannot be loaded, so that junit test IPFIXAggregatorTest.`dataAndDataTemplate` test passes with the provided pcap file.

## Steps To Reproduce
1. Load the attached pcap file and run the unit test `dataAndDataTemplate` to reproduce the error.

## Error Information

```
java.lang.IndexOutOfBoundsException: readerIndex(348) + length(8) exceeds writerIndex(352): UnpooledHeapByteBuf(ridx: 348, widx: 352, cap: 352/352)

	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1477)
	at io.netty.buffer.AbstractByteBuf.readLong(AbstractByteBuf.java:836)
	at org.graylog.integrations.ipfix.IpfixParser.parseDataSet(IpfixParser.java:478)
	at org.graylog.integrations.ipfix.codecs.IpfixCodec.lambda$decodeMessages$3(IpfixCodec.java:206)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at org.graylog.integrations.ipfix.codecs.IpfixCodec.decodeMessages(IpfixCodec.java:212)
	at org.graylog.integrations.ipfix.codecs.IpfixAggregatorTest.lambda$dataAndDataTemplate$0(IpfixAggregatorTest.java:165)
	at io.pkts.Pcap.loop(Pcap.java:85)
	at org.graylog.integrations.ipfix.codecs.IpfixAggregatorTest.dataAndDataTemplate(IpfixAggregatorTest.java:153)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

[Closed By the Issue](https://github.com/Graylog2/graylog-plugin-integrations/pull/378)

## Environment

- Graylog Version:
- Elasticsearch Version:
- MongoDB Version:
- Browser Version:

**Comment by lingpri** [comment]
[ixflow (2).pcap.zip](https://github.com/Graylog2/graylog-plugin-integrations/files/4079958/ixflow.2.pcap.zip)

**Comment by lingpri** [comment]
[ixia-ipfix2.zip](https://github.com/Graylog2/graylog-plugin-integrations/files/4079966/ixia-ipfix2.zip)

**Comment by lingpri** [comment]
![xia1](https://user-images.githubusercontent.com/6947168/72659162-1854da80-3981-11ea-95b7-c0516f47a0cc.png)

**Comment by lingpri** [comment]
![xia](https://user-images.githubusercontent.com/6947168/72659190-74b7fa00-3981-11ea-8ed7-222c553738e9.png)

**Comment by lingpri** [comment]
IPFIX DataTypes

![ipfix_dataTypes](https://user-images.githubusercontent.com/6947168/72761092-84615980-3ba0-11ea-823d-f8cf5aba95ed.png)

**Comment by lingpri** [comment]
![ipfix_sets_versions](https://user-images.githubusercontent.com/6947168/72761271-1a957f80-3ba1-11ea-99d5-53042e272ba8.png)

**Comment by lingpri** [comment]
closed by #378
